src/ngx_http_lua_exitworkerby.c

```diff
@@ -0,0 +1,101 @@
+
+/*
+ * Copyright (C) Yichun Zhang (agentzh)
+ */
+
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_exitworkerby.h"
+#include "ngx_http_lua_util.h"
+
+
+void
+ngx_http_lua_exit_worker(ngx_cycle_t *cycle)
+{
+    ngx_http_lua_main_conf_t    *lmcf;
+    ngx_connection_t            *c = NULL;
+    ngx_http_request_t          *r = NULL;
+    ngx_http_lua_ctx_t          *ctx;
+    ngx_http_conf_ctx_t         *conf_ctx;
+
+    lmcf = ngx_http_cycle_get_module_main_conf(cycle, ngx_http_lua_module);
+    if (lmcf == NULL
+        || lmcf->exit_worker_handler == NULL
+        || lmcf->lua == NULL)
```

Comment:
Should we also skip the `exit_worker_by_lua*` in non-worker processes?

Reply:
> Should we also skip the `exit_worker_by_lua*` in non-worker processes?

Thanks for review, is it the `also` mean `cache manager and cache loader processes`?

Reply:
@rainingmaster Agreed. See also the conditions for skipping `init_worker_by_lua`: https://github.com/openresty/lua-nginx-module/blob/master/src/ngx_http_lua_initworkerby.c#L51-L68. I think some tests are in order for this condition as well.

Reply:
Done, thank you both, but I don't sure the test case is cover enough or not.